pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,35 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional OpenShift components are specified in the
+// install-config to be included in the installation.
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allCapabilitySets := sets.NewString()
+	allAvailableCapabilities := sets.NewString()
+	// Create a map and slice of *all* available sets and capabilities across all capability sets
+	for baselineSet, capabilities := range configv1.ClusterVersionCapabilitySets {
+		if !allCapabilitySets.Has(string(baselineSet)) {
+			allCapabilitySets.Insert(string(baselineSet))
+		}
+		for _, capability := range capabilities {
+			if !allAvailableCapabilities.Has(string(capability)) {
+				allAvailableCapabilities.Insert(string(capability))
+			}
+		}
+	}
+
+	if len(c.BaselineCapabilitySet) > 0 && !allCapabilitySets.Has(string(c.BaselineCapabilitySet)) {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("baselineCapabilitySet"), c.BaselineCapabilitySet, allCapabilitySets.List()))
+	}
+
+	// Check to see the validity of additionalEnabledCapabilities specified by the user
+	for _, capability := range c.AdditionalEnabledCapabilities {
+		if len(capability) > 0 && !allAvailableCapabilities.Has(string(capability)) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("additionalEnabledCapabilities"), capability, allAvailableCapabilities.List()))
```

Comment:
1. An empty capability string should be rejected not accepted.
2. Include the index of the offending capability in the field error.
```suggestion
	for i, capability := range c.AdditionalEnabledCapabilities {
		if !allAvailableCapabilities.Has(string(capability)) {
			allErrs = append(allErrs, field.NotSupported(fldPath.Child("additionalEnabledCapabilities").Index(i), capability, allAvailableCapabilities.List()))
```